fix: rounding issue with tick calculation

NoelStephensUnity opened the pull request:
Fixing rounding issue with tick calculation.

[MTT-2063](https://jira.unity3d.com/browse/MTT-2063)

### PR Checklist
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.

## Changelog
### com.unity.netcode.gameobjects
Fixed network tick value sometimes being duplicated or skipped.

## Testing and Documentation
* Includes unit test: NetworkTickAdvanceTest
* No documentation changes or additions were necessary.

NoelStephensUnity committed:
fix

Fixing rounding issue with tick calculation.

NoelStephensUnity committed:
test

Added tick increment test to verify that the tick value no longer has the casted integer rounding error.

NoelStephensUnity committed:
Merge branch 'develop' into fix/ticksystem-duplicated-or-skipped-tick-value

NoelStephensUnity committed:
Update CHANGELOG.md

NoelStephensUnity committed:
fix

Had to do switch TestToFixedTime over to an approximated comparison due to the delta variances in the test values being used with the updated tick rounding fix.

NoelStephensUnity committed:
fix tests

Fixing some additional tests that appear to be taking into account the tick not being rounded.

NoelStephensUnity committed:
fix

Just testing a silly fix.

NoelStephensUnity committed:
Revert "fix tests"

This reverts commit 74aa521e4f01234a010ad825745e3412061d7a3c.

NoelStephensUnity committed:
fix

Slight modification and reverted some test changes back to original values.

NoelStephensUnity committed:
fix

increasing maximum rounding threshold to 0.999999999999

NoelStephensUnity committed:
refactor

Just trying to condense the code a little and remove the Math.Round

LukeStampfli commented on com.unity.netcode.gameobjects/Runtime/Timing/NetworkTime.cs:
```diff
@@ -113,7 +113,12 @@ public NetworkTime TimeTicksAgo(int ticks)
-            m_CachedTick = (int)d;
+            var truncated = Math.Truncate(d);
+            m_CachedTick = (int)truncated;
+            if ((d - truncated) >= 0.999999999999)
```
```suggestion
            // This check is needed due to double division imprecision of large numbers
            if ((d - truncated) >= 0.999999999999)
```
Just a small suggestion to document this for the future

NoelStephensUnity committed:
refactor

Removing the Math.Truncate stuff...
I think this is as condensed as it can possibly get.

NoelStephensUnity committed:
style

Adding comment as to what the precision check is for.
